src/main/java/io/kestra/plugin/jms/JMSConsumer.java

```diff
@@ -0,0 +1,220 @@
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.FileSerde;
+import io.kestra.core.utils.Rethrow;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.net.URI;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A Kestra task to consume messages from a JMS-compliant message broker.
+ * This task connects to a broker, consumes messages until a specified limit
+ * is reached, and stores them in Kestra's internal storage.
+ */
+@SuperBuilder
+@ToString
+@EqualsAndHashCode(callSuper = true)
+@Getter
+@NoArgsConstructor
+@Schema(title = "Consume messages from a JMS queue or topic.", description = "It is recommended to set `maxWaitTimeout` or `maxMessages`.")
+@Plugin(examples = {
+        @Example(
+                full = true,
+                title = "Consume 100 Messages from a JMS Queue",
+                code = {
+                        "id: jms_consume",
+                        "namespace: company.team",
+                        "",
+                        "tasks:",
+                        "  - id: consume_from_queue",
+                        "    type: io.kestra.plugin.jms.JMSConsumer",
+                        "    connectionFactoryConfig:",
+                        "      type: DIRECT",
+                        "      providerJarPaths: kestra:///jms/activemq-client.jar",
+                        "      connectionFactoryClass: org.apache.activemq.ActiveMQConnectionFactory",
+                        "      username: admin",
+                        "      password: \"{{ secret('AMQ_PASSWORD') }}\"",
+                        "    destination:",
+                        "      name: my-queue",
+                        "      destinationType: QUEUE",
+                        "    maxMessages: 100",
+                        "    maxWaitTimeout: 5000",
+                }
+        )
+})
+
+public class JMSConsumer extends AbstractJmsTask implements RunnableTask<JMSConsumer.Output> {
+
+    // NOTE: Using @PluginProperty instead of Property<JMSDestination> wrapper.
+    // Nested configuration objects with @PluginProperty fields don't deserialize correctly
+    // when wrapped in Property<>. Other Kestra messaging plugins (AMQP, Solace) avoid nested
+    // config objects entirely, using flat Property<String> fields instead.
+    @PluginProperty
+    @NotNull
+    @Schema(title = "The destination to consume messages from.")
+    private JMSDestination destination;
+
+    @PluginProperty(dynamic = true)
+    @Schema(
+            title = "Message selector to only consume specific messages.",
+            description = "A JMS message selector expression to filter messages. Uses SQL-92 syntax (e.g., \"JMSPriority > 5 AND type = 'order'\")."
+    )
+    private String messageSelector;
```

Comment:
+ render this property below

```suggestion
    @Schema(
            title = "Message selector to only consume specific messages.",
            description = "A JMS message selector expression to filter messages. Uses SQL-92 syntax (e.g., \"JMSPriority > 5 AND type = 'order'\")."
    )
    private Property<String> messageSelector;
```